javalib/src/main/scala/java/lang/UnixProcess.scala

```diff
@@ -0,0 +1,296 @@
+
+import java.io.{File, InputStream, IOException, OutputStream}
+import java.util.concurrent.TimeUnit
+
+import scala.annotation.tailrec
+import scala.scalanative.native.{signal => sig, errno => err, _}, sig._,
+err.errno
+import scala.scalanative.posix.{fcntl, sys, unistd, errno => e}, sys.time._,
+e.ETIMEDOUT
+
+import UnixProcess._
+
+private[lang] class UnixProcess private (
+    pid: CInt,
+    _inputStream: PipeIO.Stream,
+    _errorStream: PipeIO.Stream,
+    _outputStream: OutputStream
+) extends Process {
+  override def destroy(): Unit = kill(pid, 9)
+
+  override def destroyForcibly(): Process = {
+    destroy()
+    this
+  }
+
+  override def exitValue(): scala.Int = locked { _ =>
+    _exitValue match {
+      case -1 =>
+        throw new IllegalThreadStateException(
+          s"Process $pid has not exited yet")
+      case v => v
+    }
+  }
+
+  override def getErrorStream(): InputStream = _errorStream
+
+  override def getInputStream(): InputStream = _inputStream
+
+  override def getOutputStream(): OutputStream = _outputStream
+
+  override def isAlive(): scala.Boolean = locked(_ => _exitValue == -1)
+
+  override def toString = s"UnixProcess($pid)"
+
+  override def waitFor(): scala.Int = locked { m =>
+    _exitValue match {
+      case -1 =>
+        waitImpl(() => UnixProcess.waitFor(m, this, null))
+        _exitValue
+      case v => v
+    }
+  }
+
+  override def waitFor(timeout: scala.Long, unit: TimeUnit): scala.Boolean =
+    locked {
+      case m =>
+        _exitValue match {
+          case -1 =>
+            val ts = stackalloc[timespec]
+            val tv = stackalloc[timeval]
+            throwOnError(gettimeofday(tv, null), "Failed to set time of day.")
+            val nsec = unit.toNanos(timeout) + TimeUnit.MICROSECONDS.toNanos(
+              !tv._2.cast[Ptr[CInt]])
+            val sec = TimeUnit.NANOSECONDS.toSeconds(nsec)
+            !ts._1 = !tv._1 + sec
+            !ts._2 = if (sec > 0) nsec - TimeUnit.SECONDS.toNanos(sec) else nsec
+            waitImpl(() => UnixProcess.waitFor(m, this, ts)) == 0
+          case _ => true
+        }
+    }
+
+  @inline private def waitImpl(f: () => Int) = {
+    var res = 1
+    do res = f() while (if (res == 0) _exitValue == -1 else res != ETIMEDOUT)
+    res
+  }
+
+  private[this] var _exitValue                  = -1
+  private[this] var _cond: Ptr[Ptr[scala.Byte]] = null
+  private def allocCond() = {
+    if (_cond == null)
+      _cond = stdlib.malloc(sizeof[Ptr[scala.Byte]]).cast[Ptr[Ptr[scala.Byte]]]
+    _cond
+  }
+  private def freeCond() = _cond = {
+    stdlib.free(_cond.cast[Ptr[scala.Byte]]); null
+  }
+  private def signalCond() = if (_cond != null) {
+    UnixProcess.signalCond(!_cond)
+    freeCond()
+  }
+  private def setExitValue(value: CInt) = {
+    _exitValue = value
+    _inputStream.drain()
+    _errorStream.drain()
+    _outputStream.close()
+  }
+}
+
+object UnixProcess {
+  @extern
+  private[this] object ProcessMonitor {
```

Comment:
The methods that just bind to posix should go to `scala.scalanative.posix.pthread`. The others which are "runtime" code for process implementation needs to go to `scala.scalanative.runtime.ProcessMonitor`.

Reply:
Also needs `@link("pthread")`.

Reply:
Part of the reason that I didn't put things in scala.scalanative.posix.pthread was because I knew that https://github.com/scala-native/scala-native/pull/1068/ was in flight. I didn't want to create conflicts for that PR since its so big. Also, I figured that after thread support was merged, the implementation could be redone in scala native which would remove the need for these bindings and process_monitor.c. At any rate, the pthread.scala file is not on the master branch, so I'd have to add it. I can see three options:

1) Create a new file with the specific bindings I need
2) Copy the file from 1068
3) Leave things as is

I'm happy to do any of the 3.

Reply:
Denys is on vacation this week but I would go with option 2. If your change goes first #1068 can rebase on top of your changes or visa versa. That also helps makes review fixes easier.